Source/Scene/GroundPrimitiveBatch.js

```diff
@@ -0,0 +1,997 @@
+
+        var needToRebatch = false;
+        var colorCounts = {};
+
+        for (var i = 0; i < length; ++i) {
+            var color = batchedIndices[i].color;
+            var rgba = color.toRgba();
+            if (defined(colorCounts[rgba])) {
+                needToRebatch = true;
+                break;
+            } else {
+                colorCounts[rgba] = true;
+            }
+        }
+
+        if (!needToRebatch) {
+            primitive._batchDirty = false;
+            return false;
+        }
+
+        batchedIndices.sort(compareColors);
+
+        var newIndices = new primitive._indices.constructor(primitive._indices.length);
+
+        var current = batchedIndices.pop();
+        var newBatchedIndices = [current];
+
+        var currentOffset = copyIndices(primitive._indices, newIndices, 0, primitive._indexOffsets, primitive._indexCounts, current.batchIds);
+
+        current.offset = 0;
+        current.count = currentOffset;
+
+        while (batchedIndices.length > 0) {
+            var next = batchedIndices.pop();
+            if (Color.equals(next.color, current.color)) {
+                currentOffset = copyIndices(primitive._indices, newIndices, currentOffset, primitive._indexOffsets, primitive._indexCounts, next.batchIds);
+                current.batchIds = current.batchIds.concat(next.batchIds);
+                current.count = currentOffset - current.offset;
+            } else {
+                var offset = currentOffset;
+                currentOffset = copyIndices(primitive._indices, newIndices, currentOffset, primitive._indexOffsets, primitive._indexCounts, next.batchIds);
+
+                next.offset = offset;
+                next.count = currentOffset - offset;
+                newBatchedIndices.push(next);
+                current = next;
+            }
+        }
+
+        primitive._va.indexBuffer.copyFromArrayView(newIndices);
+
+        primitive._indices = newIndices;
+        primitive._batchedIndices = newBatchedIndices;
+
+        primitive._batchDirty = false;
+        primitive._pickCommandsDirty = true;
+        return true;
+    }
+
+    function createColorCommands(primitive) {
+        if (defined(primitive._commands) && !rebatchCommands(primitive) && primitive._commands.length / 3 === primitive._batchedIndices.length) {
+            return;
+        }
+
+        var batchedIndices = primitive._batchedIndices;
+        var length = batchedIndices.length * 3;
+
+        var commands = primitive._commands;
+        commands.length = length;
+
+        var vertexArray = primitive._va;
+        var uniformMap = primitive._batchTable.getUniformMapCallback()(primitive._uniformMap);
+        var bv = primitive._boundingVolume;
+
+        for (var j = 0; j < length; j += 3) {
+            var offset = batchedIndices[j / 3].offset;
+            var count = batchedIndices[j / 3].count;
+
+            // stencil preload command
+            var command = commands[j];
```

Comment:
Can we avoid reusing the same `command` variable here for three different commands?  Just name each one based on the pass.